cartography/intel/aws/cloudtrail_management_events.py

```diff
@@ -0,0 +1,553 @@
+    """
+    role_assumptions = []
+
+    logger.info(
+        f"Transforming {len(events)} CloudTrail events to role assumptions for region '{region}'"
+    )
+
+    for event in events:
+        try:
+            assumption = _extract_role_assumption_from_event(
+                event, region, current_aws_account_id
+            )
+            if assumption:
+                role_assumptions.append(assumption)
+        except Exception as e:
+            logger.warning(
+                f"Failed to transform CloudTrail event {event.get('EventId', 'unknown')}: {str(e)}"
+            )
+            continue
+
+    logger.info(
+        f"Successfully transformed {len(role_assumptions)} role assumptions from {len(events)} events"
+    )
+    return role_assumptions
+
+
+@timeit
+def load_role_assumptions(
+    neo4j_session: neo4j.Session,
+    role_assumptions: List[Dict[str, Any]],
+    region: str,
+    current_aws_account_id: str,
+    aws_update_tag: int,
+) -> None:
+    """
+    Load role assumption relationships into Neo4j using MatchLink pattern.
+
+    Creates direct ASSUMED_ROLE relationships with aggregated properties:
+    (AWSUser|AWSRole|AWSPrincipal)-[:ASSUMED_ROLE {lastused, times_used, first_seen, last_seen}]->(AWSRole)
+
+    Assumes that both source principals and destination roles already exist in the graph.
+
+    :type neo4j_session: neo4j.Session
+    :param neo4j_session: The Neo4j session to use for database operations
+    :type role_assumptions: List[Dict[str, Any]]
+    :param role_assumptions: List of role assumption records from transform function
+    :type region: str
+    :param region: The AWS region where events were retrieved from
+    :type current_aws_account_id: str
+    :param current_aws_account_id: The AWS account ID being synced
+    :type aws_update_tag: int
+    :param aws_update_tag: Timestamp tag for tracking data freshness
+    :rtype: None
+    """
+    if not role_assumptions:
+        logger.info("No role assumption events to load")
+        return
+
+    # Aggregate role assumptions by (source, destination) pairs
+    aggregated_assumptions = _aggregate_role_assumptions(role_assumptions)
+
+    logger.info(
+        f"Loading {len(aggregated_assumptions)} aggregated role assumption relationships"
+    )
+
+    # Use MatchLink to create relationships between existing nodes
+    matchlink_schema = AssumedRoleMatchLink()
+
+    load_matchlinks(
+        neo4j_session,
+        matchlink_schema,
+        aggregated_assumptions,
+        lastupdated=aws_update_tag,
+        _sub_resource_label="AWSAccount",
+        _sub_resource_id=current_aws_account_id,
+    )
+
+    logger.info(
+        f"Successfully loaded {len(aggregated_assumptions)} role assumption relationships"
+    )
+
+
+def _aggregate_role_assumptions(
+    role_assumptions: List[Dict[str, Any]],
+) -> List[Dict[str, Any]]:
+    """
+    Aggregate role assumption events by (source_principal, destination_principal) pairs.
```

Comment:
We should think a bit more on an optimal approach here since this is likely to be a memory bottleneck